testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestAllDatatypesFromHiveConnector.java

```diff
@@ -152,6 +163,68 @@ public void testSelectAllDatatypesTextFile()
                         "kot binarny".getBytes(UTF_8)));
     }
 
+    @Requires(OpenXJsonRequirements.class)
+    @Test(groups = SMOKE)
+    public void testSelectAllDatatypesOpenX()
+    {
+        String tableName = ALL_HIVE_SIMPLE_TYPES_OPENX.getName();
+
+        assertThat(query("SHOW COLUMNS FROM " + tableName).project(1, 2)).containsExactly(
+                row("c_tinyint", "tinyint"),
+                row("c_smallint", "smallint"),
+                row("c_int", "integer"),
+                row("c_bigint", "bigint"),
+                row("c_float", "real"),
+                row("c_double", "double"),
+                row("c_decimal", "decimal(10,0)"),
+                row("c_decimal_short", "decimal(10,5)"),
+                row("c_decimal_full", "decimal(38,18)"),
+                row("c_timestamp", "timestamp(3)"),
+                row("c_date", "date"),
+                row("c_string", "varchar"),
+                row("c_varchar", "varchar(10)"),
+                row("c_char", "char(10)"),
+                row("c_boolean", "boolean"),
+                row("c_binary", "varbinary"));
+        QueryResult queryResult = query(format("SELECT * FROM %s", tableName));
+
+        assertThat(queryResult).hasColumns(
+                TINYINT,
+                SMALLINT,
+                INTEGER,
+                BIGINT,
+                REAL,
+                DOUBLE,
+                DECIMAL,
+                DECIMAL,
+                DECIMAL,
+                TIMESTAMP,
+                DATE,
+                VARCHAR,
+                VARCHAR,
+                CHAR,
+                BOOLEAN,
+                VARBINARY);
+        assertThat(queryResult).containsOnly(
```

Comment:
does it test edge cases that are specifically handled by openx?

Reply:
I think I still need to add a couple test cases for this serde. It seems I left some features untested.